### PR TITLE
Donation view first version without backend

### PR DESCRIPTION
**Pull request by mateodaza**

### CODE REVIEW

File: src/apollo/apolloClient.ts [hunk 1]
```diff
@@ -1,25 +1,131 @@
...
-    query: {
-      fetchPolicy: 'network-only'
+export const APOLLO_STATE_PROP_NAME = '__APOLLO_STATE__'
+
+function createApolloClient() {
+  // Declare variable to store authToken
+  let token: any
+  const appUser = getLocalStorageUserLabel()
+
+  const httpLink = createUploadLink({
```
Comment (RamRamez) [inline src/apollo/apolloClient.ts:22]: I think we don't need to use createUploadLink. Use `createHttpLink` instead

File: src/apollo/apolloClient.ts [hunk 2]
```diff
@@ -1,25 +1,131 @@
...
+    uri: process.env.NEXT_PUBLIC_APOLLO_SERVER,
```
Comment (RamRamez) [inline src/apollo/apolloClient.ts]: You can use `config.ts` here. No need to pass env.
We save all needed variables in the config file.

File: src/apollo/auth.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+export const isBrowser = () => typeof window !== 'undefined'
```
Comment (RamRamez) [inline src/apollo/auth.js]: You can remove this file (`auth.js`). I'm saving user data on the UserProvider component.

File: src/apollo/gql/gqlProjects.ts [hunk 4]
```diff
@@ -73,10 +73,13 @@ export const FETCH_PROJECT_BY_SLUG = gql`
       slug
       description
       verified
-      traceCampaignId
       reactions {
         userId
       }
+      users {
```
Comment (RamRamez) [inline src/apollo/gql/gqlProjects.ts]: Is it necessary to use `users`? why not using `adminUser`?
As I remember, `users` was null for some reason for some of the projects.

File: src/components/project-card/ProjectCard.tsx [hunk 5]
```diff
@@ -35,10 +42,19 @@ const ProjectCard = (props: IProjectCard) => {
 
   const router = useRouter()
 
-  const name = adminUser.name
+  // const projectImage = () => {
+  //   if (isNoImg(image)) return <NoImg />
+  //   return <Img src={image} alt='project image' />
+  // }
+
+  const name = users?.length > 0 && users[0].name
```
Comment (RamRamez) [inline src/components/project-card/ProjectCard.tsx]: why not using `adminUser` here?
This way project owner is not shown on project cards anymore.

File: src/components/project-card/ProjectCard.tsx [hunk 6]
```diff
@@ -87,6 +103,18 @@ const HoverButtons = styled.div`
     width: 100%;
   }
 `
+const NoImg = styled.div`
```
Comment (RamRamez) [inline src/components/project-card/ProjectCard.tsx:99]: you can remove them.

File: src/components/project-card/ProjectCardAlt.tsx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    <Wrapper>
+      <Wrapper2>
+        <ImagePlaceholder>{projectImage()}</ImagePlaceholder>
+        <ProjectCardBadges cardWidth={cardWidth} likes={reactions.length} verified={verified} />
+        <CardBody>
+          <Title>{title}</Title>
+          {name && <Author>{name}</Author>}
+          <Description>{htmlToText(description)}</Description>
+          <Captions>
+            <Caption>Raised: $200</Caption>
```
Comment (RamRamez) [inline src/components/project-card/ProjectCardAlt.tsx:36]: You can use `totalDonations` for this.

File: src/components/project-card/ProjectCardAlt.tsx [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+            <Caption>Last updated: 5 days ago</Caption>
```
Comment (RamRamez) [inline src/components/project-card/ProjectCardAlt.tsx:37]: You can use `updatedAt` for this.

File: src/components/project-card/ProjectCardAlt.tsx [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+const ImagePlaceholder = styled.div`
+  height: ${imgHeight};
+  width: 100%;
+  position: relative;
+  overflow: hidden;
+  border-radius: 16px;
+`
+
+const Wrapper2 = styled.div`
```
Comment (RamRamez) [inline src/components/project-card/ProjectCardAlt.tsx:95]: `Wrapper2` was used for hover state for project cards. You can remove it here.

File: src/components/views/donate/CryptoDonation.tsx [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+  height: 54px;
+`
+const SearchBarContainer = styled.div`
+  height: 54px;
+  width: 65%;
+  border: 2px solid ${Gray_300};
+`
+const XDaiContainer = styled.div`
+  display: flex;
+  flex: row;
```
Comment (RamRamez) [inline src/components/views/donate/CryptoDonation.tsx:173]: `flex-direction` ?

File: src/components/views/donate/DonateIndex.tsx [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+} from '../../styled-components/Colors'
+import { BigArc } from '../../styled-components/Arc'
+import ArrowLeft from '../../../../public/images/arrow_left.svg'
+import RadioOnIcon from '../../../../public/images/radio_on.svg'
+import RadioOffIcon from '../../../../public/images/radio_off.svg'
+
+const CRYPTO_DONATION = 'Cryptocurrency'
+const FIAT_DONATION = 'Credit Card'
+
+const ProjectsIndex = (props: IProjectBySlug) => {
```
Comment (RamRamez) [inline src/components/views/donate/DonateIndex.tsx:34]: It should be `DonateIndex`

File: src/components/views/donate/DonateIndex.tsx [hunk 12]
```diff
@@ -0,0 +1,224 @@
...
+          <Link_Big>Your account</Link_Big>
+        </Options>
+      </SucceessContainer>
+    )
+  }
+
+  return (
+    <>
+      <BigArc />
+      <Wrapper>
```
Comment (RamRamez) [inline src/components/views/donate/DonateIndex.tsx:104]: It has `border-radius` on design

File: pages/project/[slug].tsx [hunk 13]
```diff
@@ -27,13 +28,15 @@ export async function getServerSideProps(props: { query: { slug: string } }) {
...
+    fetchPolicy: 'no-cache',
   })
+  const project = data.projectBySlug
 
   return {
     props: {
-      project: data.projectBySlug
-    }
+      project,
+      admin: project.adminUser,
```
Comment (RamRamez) [inline pages/project/[slug].tsx]: Is it used anywhere?

File: src/components/project-card/ProjectCard.tsx [hunk 14]
```diff
@@ -6,7 +6,14 @@ import { Body_P, H6, Caption } from '../styled-components/Typography'
 import { Gray_900, Pinky_500, Primary_Deep_500 } from '../styled-components/Colors'
 import ProjectCardBadges from './ProjectCardBadges'
 import { IProject } from '../../types/types'
-import { htmlToText, slugToProjectDonate, slugToProjectView } from '../../lib/helpers'
+import {
+  htmlToText,
+  isNoImg,
```
Comment (RamRamez) [inline src/components/project-card/ProjectCard.tsx]: not used.

File: src/components/project-card/ProjectCardAlt.tsx [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+
+interface IProjectCard {
+  project: IProject
+}
+
+const ProjectCard = (props: IProjectCard) => {
+  const [rndColor, setRndColor] = useState(noImgColor)
+  const { title, description, image, verified, reactions, users } = props.project
+
+  const projectImage = () => {
```
Comment (RamRamez) [inline src/components/project-card/ProjectCardAlt.tsx]: you can use `ProjectCardImage` instead.

File: src/components/project-card/ProjectCardAlt.tsx [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+    if (isNoImg(image)) return <NoImg rndColor={rndColor} />
+    return <Img src={image} alt='project image' />
+  }
+
+  const name = users?.length > 0 && users[0].name
+
+  return (
+    <Wrapper>
+      <Wrapper2>
+        <ImagePlaceholder>{projectImage()}</ImagePlaceholder>
```
Comment (RamRamez) [inline src/components/project-card/ProjectCardAlt.tsx]: `<ImagePlaceholder>
          <ProjectCardImage image={image} cardWidth={cardWidth} cardRadius={cardRadius} />
        </ImagePlaceholder>`

File: src/components/views/donate/DonateIndex.tsx [hunk 17]
```diff
@@ -0,0 +1,224 @@
...
+          <Right>
+            {isSuccess ? (
+              <SuccessView />
+            ) : (
+              <>
+                <H4>Donate With</H4>
+                <TypeSelection />
+              </>
+            )}
+          </Right>
```
Comment (RamRamez) [inline src/components/views/donate/DonateIndex.tsx:123]: please also add share buttons on the design

**Review by RamRamez — CHANGES_REQUESTED** [review]
Thanks @mateodaza. It's great just some minor fixes.

File: src/apollo/apolloClient.ts (hunk 1 above)
Comment (mateodaza) [reply src/apollo/apolloClient.ts:22]: We need uploadLink to be able to send images through graphql, it's gonna be needed when we implement create/edit projects

**Pull request merged**